src/test/unit/algorithms/ConnectionsTest.cpp

```diff
@@ -463,6 +463,13 @@ TEST(ConnectionsTest, testRaisePermanencesToThreshold) {
                                                       synPermBelowStimulusInc);
     }
   }
+
+  // check empty segment (with no synapse data) 
+  auto emptySegment = con.createSegment(0);
+  auto synapses = con.synapsesForSegment(emptySegment);
+  NTA_CHECK(synapses.empty()) << "We want to create a Segment with none synapses";
```

Comment:
can this normally occur? Or better, can we make an asumption (and change, as it happens now) that Connections never create a Segment w/o synapses?